### PR TITLE
Added new FeralFileArtworkV3 contract

### PR DESCRIPTION
**Pull request by tienngovan**

### CODE REVIEW

**Review by lemonlatte — CHANGES_REQUESTED** [review]
Could you update the indent use 2 space?

For `migrations/3_feralfile_exhibition_v2.js`, it seems only indent changes?

File: contracts/FeralfileArtworkV3.sol [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+        receiver = royaltyPayoutAddress;
+
+        royaltyAmount =
+            (salePrice * secondarySaleRoyaltyBPS) /
+            MAX_ROYALITY_BPS;
+    }
+
+    /// @notice isValidRequest validates a message by ecrecover to ensure
+    //          it is signed by either the contract owner or a trustee
+    //          who is set by the contract owner.
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]: The signature comes from the token owner, the token owner might not be a trustee.
The scenario here is the trustee execute the transfer base on the consent of the token owner.

File: contracts/FeralfileArtworkV3.sol [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+        uint8 v_,
+        uint256 tokenID_
+    ) internal view returns (bool authorized) {
+        address signer = ECDSA.recover(
+            ECDSA.toEthSignedMessageHash(message_),
+            v_,
+            r_,
+            s_
+        );
+        authorized =
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]: It supposes to validate the signer is `transferParam_.from`.

**Review by lemonlatte — CHANGES_REQUESTED** [review]
The functions are modularized very well here.

File: contracts/FeralfileArtworkV3.sol [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        require(
+            artworks[artworkID_].editionSize > 0,
+            "FeralfileExhibitionV3: artwork is not found"
+        );
+        /// @notice The range of editionNumber should be between 0 (AP) ~ artwork.editionSize
+        require(
+            editionNumber_ <= artworks[artworkID_].editionSize,
+            "FeralfileExhibitionV3: edition number exceed the edition size of the artwork"
+        );
+        require(owner_ != address(0), "invalid owner address");
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol:400]:
```suggestion
        require(owner_ != address(0), "invalid owner address");
        require(artist_ != address(0), "invalid artist address");
```
If we do not accept artist in zero address.

**Review by lemonlatte — CHANGES_REQUESTED** [review]

File: contracts/FeralfileArtworkV3.sol (hunk 2 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol]: Fixed, changed verify signer is transferParam_.from

File: contracts/FeralfileArtworkV3.sol (hunk 3 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol:400]: Added

**Review by lemonlatte — APPROVED** [review]
LGTM

**Pull request merged**